### ISSUE TITLE
Test multiple MySQL and MariaDB versions

### ISSUE THREAD

**Issue by zuiderkwast** [mysql-otp/mysql-otp#99]
Use Travis matrix. See e.g. https://github.com/xerions/mariaex/blob/master/.travis.yml or https://github.com/mysqljs/mysql/blob/master/.travis.yml

**Comment by michaellenaghan** [comment]
Another option—as in "in addition to" rather than "in place of"—would be to use Docker the way MyXQL does [here](https://github.com/elixir-ecto/myxql/blob/master/ci.sh).

Notice that with their Docker approach you can select the version you want to test by using an environment variable, e.g.:

```
DB=mysql:5.5 ./ci.sh ...
DB=mysql:5.6 ./ci.sh ...
DB=mysql:5.7 ./ci.sh ...
```

**Comment by zuiderkwast** [comment]
Nice! For Travis we can definitely go for this!

As long as it's still possible to run the tests locally without Docker.

**Comment by michaellenaghan** [comment]
They just use Docker to run MySQL; tests run locally, but against a Dockerized MySQL. Very simple idea, very simple implementation.

Running against both Dockerized and not-Dockerized MySQL would be trivially easy. Off the top of my head I can think of two issues. First, you'd want to parameterize some connection info in your tests; for example, I don't think you'd want Dockerized MySQL running on port 3306. Second, you'd want to sometimes automatically create the MySQL users you currently ask people to create themselves.

**Comment by zuiderkwast** [comment]
Yes I know. I just want it to be possible to run the tests using a locally installed MySQL (without docker) as well.

> I don't think you'd want Dockerized MySQL running on port 3306

Why not?

> Second, you'd want to sometimes automatically create the MySQL users you currently ask people to create themselves.

Sure, feel free to change this!

We have one test case for connecting using unix socket. That only works if the database is running locally. I don't know if this works with docker.

**Comment by michaellenaghan** [comment]
>> I don't think you'd want Dockerized MySQL running on port 3306

> Why not?

I run MySQL locally. I wouldn't want to have to stop it just so that I could run mysql-otp tests. Not only that, but by making the port configurable you could in theory test different MySQL versions in parallel.

**Comment by zuiderkwast** [comment]
Fixed in #157 

Future work:
* Add MariaDB 10.4 and 10.5
* Generate combined code coverage from the different test runs
